source/common/common/callback_impl.h

```diff
@@ -46,24 +47,21 @@ template <typename... CallbackArgs> class CallbackManager {
     CallbackHolder(CallbackManager& parent, Callback cb) : parent_(parent), cb_(cb) {}
 
     // CallbackHandle
-    void remove() override { parent_.remove(this); }
+    void remove() override { parent_.remove(it_); }
```

Comment:
So, I think the safety of this depends on:

"Adding, removing and moving the elements within the list or across several lists does not invalidate the iterators or references. An iterator is invalidated only when the corresponding element is deleted."

in https://en.cppreference.com/w/cpp/container/list. I.e. that this is a `std::list`. I'd add some comments to the call sites here and where `it_` is initialized, as well as to where `callbacks_` is defined, to make sure that this is clear as an invariant.

Reply:
Yes, it is guaranteed by C++ standard that list iterator won't be invalidated by list operation.